### PR TITLE
cmake: improve building optimization & flags for heap analysis

### PR DESCRIPTION
**Pull request by MatusKysel**
## Purpose

This PR consist of 2 things:

- most of the changes are improving building optimization
- I have also add linking with gperftools for heapanalysis

### CODE REVIEW

File: submodules/CMakeLists.txt [hunk 1]
```diff
@@ -94,7 +94,7 @@ endfunction()
 # Add taraxa-vdf
 set(VDF_LIBRARY_COMMAND "\
  ${CMAKE_MAKE_PROGRAM}\
- CPPFLAGS=\"-I./include -I${CONAN_INCLUDE_DIRS_OPENSSL} -I${CONAN_INCLUDE_DIRS_GMP} -I${CONAN_INCLUDE_DIRS_MPFR} -std=c++17\"\
+ CPPFLAGS=\"-I./include -I${CONAN_INCLUDE_DIRS_OPENSSL} -I${CONAN_INCLUDE_DIRS_GMP} -I${CONAN_INCLUDE_DIRS_MPFR} -std=c++17 -O3\"\
```
Comment (JakubFornadel) [inline submodules/CMakeLists.txt:97]: isn't this set automatically according to CMAKE_CXX_FLAGS_RELEASE var ?

Comment (MatusKysel) [reply submodules/CMakeLists.txt:97]: not in our case, as we are using special script `setenv_for_taraxa_vdf.sh ` that is generated by this command. So we completely overwrite whatever is in CPPFLAGS

Comment (JakubFornadel) [reply submodules/CMakeLists.txt:97]: aah ok

**Review by JakubFornadel — APPROVED** [review]

**Review by justinsnapp — APPROVED** [review]
LGTM

**Pull request merged**